core/shared/src/main/scala/zio/concurrent/ConcurrentMap.scala

```diff
@@ -58,6 +119,30 @@ final class ConcurrentMap[K, V] private (private val underlying: ConcurrentHashM
   def remove(key: K, value: V): UIO[Boolean] =
     UIO(underlying.remove(key, value))
 
+  /**
+   * Removes all elements which do not satisfy the given predicate.
+   */
+  def removeIf(p: (K, V) => Boolean): UIO[Unit] =
+    UIO(
+      underlying.forEach { (k: K, v: V) =>
+        if (p(k, v)) {
+          val _ = underlying.remove(k)
```

Comment:
The only thing that I'm not sure of: can we update the collection while we are iterating over its elements?

Reply:
It seems to pass the tests. Do you have any ideas as to how we can rewrite it to make it more atomic?

Reply:
I don't think it is possible, so we'll have to live with it. Worst case we remove those methods, if they prove to create more issues than they solve.

Reply:
Okay, we'll see how it goes. Happy to remove it if needed.